test/irmin-pack/test_inode.ml

```diff
@@ -664,10 +666,145 @@ module Inode_tezos = struct
     S.Context.close t
 end
 
+module Child_ordering = struct
+  (** Tests of the relative ordering of Inode children (which can be configured
+      by the user). *)
+
+  module Step = struct
+    type t = Schema.Path.step [@@deriving irmin ~short_hash]
+
+    module Hash =
+      Irmin.Hash.Typed
+        (Schema.Hash)
+        (struct
+          type nonrec t = t
+
+          let t = t
+        end)
+
+    type nonrec hash = Hash.t [@@deriving irmin ~to_bin_string]
+
+    let hash : t -> string = fun s -> hash_to_bin_string (Hash.hash s)
+  end
+
+  module type S = Irmin_pack.Inode.Child_ordering with type step := Step.t
+
+  let make ?entries:(entries' = Irmin_tezos.Conf.entries)
+      (t : Irmin_pack.Conf.inode_child_order) : (module S) =
+    let module Conf = struct
+      include Irmin_tezos.Conf
+
+      let entries = entries'
+      let inode_child_order = t
+    end in
+    let module T = Inode_modules (Conf) (Schema) (String_contents) in
+    (module T.Inter.Child_ordering)
+
+  let check_child_index pos (module Order : S) ~reference ~step ~depth =
+    let msg =
+      Fmt.str "Short hash of child at { depth = %d; step = %S }" depth step
+    in
+    let expected = reference ~depth step in
+    let actual = Order.key step |> Order.index ~depth in
+    check_int pos ~msg ~expected actual
+
+  let check_max_depth_exception pos (module Order : S) ~step ~depth =
+    match Order.key step |> Order.index ~depth with
+    | index ->
+        Alcotest.failf ~pos
+          "Expected [Max_depth %d] to be raised, but got a computed index of \
+           %d instead"
+          depth index
+    | exception Irmin_pack.Inode.Max_depth _ -> ()
+
+  (* Get the bit at index [n] in a string: *)
+  let get_bit str n =
+    let chosen_byte = Bytes.get_uint8 (Bytes.unsafe_of_string str) (n / 8) in
+    let bit_index_in_byte = n mod 8 in
+    (* Selects only the chosen bit from our byte: *)
+    let mask = 1 lsl (7 - bit_index_in_byte) in
+    let masked_byte = chosen_byte land mask in
+    let chosen_bit = masked_byte lsr (7 - bit_index_in_byte) in
+    assert (chosen_bit = 0 || chosen_bit = 1);
+    chosen_bit
+
+  let test_seeded_hash () =
```

Comment:
These tests are great! It's maybe just nitpicking, but I think the tests for custom and seeded_hash should test for the hardcoded value of `index` (because otherwise we just check that `Order` applies `reference` correctly - which is maybe enough).

Reply:
I was considering `reference` as being a function for computing the same expected value without needing to hard-code it in the test, but I'm happy to do it the other way too! Have pushed a refactor to do this.